src/main/java/com/cvsgo/service/UserService.java

```diff
@@ -85,4 +95,52 @@ public boolean isDuplicatedEmail(String email) {
     public boolean isDuplicatedNickname(String nickname) {
         return userRepository.findByNickname(nickname).isPresent();
     }
+
+    /**
+     * 회원 팔로우를 생성한다.
+     *
+     * @param user   로그인한 사용자
+     * @param userId 팔로잉할 사용자 ID
+     * @throws NotFoundUserException         해당하는 아이디를 가진 사용자가 없는 경우
+     * @throws BadRequestUserFollowException 본인을 팔로우하는 경우
+     * @throws DuplicateUserFollowException  이미 해당하는 회원 팔로우가 존재하는 경우
+     */
+    @Transactional(rollbackFor = Exception.class)
+    public void createUserFollow(User user, Long userId) {
+        User followingUser = userRepository.findById(userId).orElseThrow(() -> NOT_FOUND_USER);
+        if (user.equals(followingUser)) {
+            throw BAD_REQUEST_USER_FOLLOW;
+        }
+
+        UserFollow userFollow = UserFollow.create(followingUser, user);
+        try {
+            userFollowRepository.save(userFollow);
+        } catch (DataIntegrityViolationException e) {
+            entityManager.clear();
+            if (Boolean.TRUE.equals(
+                userFollowRepository.existsByUserAndFollower(followingUser, user))) {
```

Comment:
Spring Data JPA의 existsBy... 메소드의 리턴 타입이 Boolean이 아니라 boolean이라서 boolean으로 받고 아래와 같이 수정하면 좋을 것 같아요!
```suggestion
            if (userFollowRepository.existsByUserAndFollower(followingUser, user)) {
```